### ISSUE TITLE
Named HTML files are not being generated

### ISSUE THREAD

**Issue by saatvikshah1994** [jekyll/jekyll#4451]
I'm currently using the [HPSTR Jekyll theme](https://github.com/mmistakes/hpstr-jekyll-theme). Until the recent update, I was able to use Named HTML files on my site's root on Github pages. However now this is returning a 404 page.

**Comment by parkr** [comment]
@saatvikshah1994 This does not clearly explain your issue. This sounds like an issue we have dealt with already, namely that `permalink` without the trailing slash would generate a file in a directory in Jekyll 2, but in Jekyll 3 it generates a file without the directory. That is #4440.

If it is not that, please clearly define what you mean. I am confused about what you mean by "Named HTML file on my site's root". If your site is open source, a link would to the repo would help.

**Comment by saatvikshah1994** [comment]
This is the [link](https://github.com/saatvikshah1994/saatvikshah1994.github.io) to my website running on github pages. [Here](https://saatvikshah1994.github.io/) is the actual website itself. 
Following the documentation [here](http://jekyllrb.com/docs/pages/), I've created a about.html in the root folder, but on trying to access it I get 404'd. The same is happening on my local system.
There's another issue I noticed which is that if you go to this [page](https://saatvikshah1994.github.io/portfolio/) and scroll below, you'll see that the markdown is malformed. On my local system this is in fact being correctly rendered and I have the  "github-pages" gem installed as well. So why is this happening?
